Handler/Sites.hs

```diff
@@ -97,6 +96,14 @@ siteForm msite = renderBootstrap3 BootstrapBasicForm $ Site
     bfl :: Text -> FieldSettings site
     bfl = bfs
 
+    baseUrlField :: Field Handler Text
+    baseUrlField = flip checkM textField $ \url -> do
```

Comment:
I realized the uniqueness constraint would result in a poor UX now because, by default, the PG_ERROR just comes through as a 500. I decided to add this validation even though it's a race condition. It improves things a lot with little effort. I may or may not add an Issue to look into doing it without the race condition (maybe rescuing the IOException and manually adding an error to the form before re-rendering).